Fs refactor

filipsch opened the pull request:
Hey @richierocks, here is the PR to intro to SQL for data science that:

- Bakes in the new versions of sqlwhat and protowhat, so you can use the new functions that were introduced in the training session. As soon as these new versions are deployed through the shared image, I'll make sure to remove this course-specific install.
- Rewrites all SCTs according to these new functions.
- Makes some small improvements to instructions and assignment text, informed by feedback messages, student submissions and my own gut feeling.

When you come across things you don't like in the:
- assignment/instructions, please change as you see fit.
- SCT or the feedback it generates, flag it and I'll have a look and see what I can do. I can't guarantee that I will be able to fix things 100%. There are still limitations to the system, as you've noticed in the training session.

I hope this moves the needle tremendously in terms of asked hint, asked solution and duped metric!

richierocks commented on chapter3.md:
```diff
@@ -1148,44 +1516,28 @@ SELECT COUNT(deathdate) * 100.0 / COUNT(*) AS percentage_dead
-temp = left_eqn.check_field('left')
-
-count_call = temp.check_field('name').has_equal_ast('Are you using the `COUNT` function for the top of your fraction?')
+Ex().check_correct(
```
I think that a common mistake here is to forget the `.0` in the number. That is, using

```sql
COUNT(deathdate) * 100 / COUNT(*)
```

instead of 

```sql
COUNT(deathdate) * 100.0 / COUNT(*)
```

In this case, the feedback message is

```sql
Check the expression of the alias expression. The checker expected to find COUNT(deathdate) * 100.0 / COUNT(*) in there.
```

This is OK, but is it easy to give a more targeted message?

filipsch commented:
@richierocks Thanks for the thorough review. I addressed your comments:

- Fixed the typo
- Did not dive deeper into AliasExpr, so the messaging is better (it's also giving away a bit more though)
- There is a specific message if people don't use `100.0`, by using `check_correct()`.

If this looks good to you, I'll merge this first thing next week and keep an eye out.

Now there's one more outstanding review for the joining data in postgresql course, unless you trust Chester's review and don't want to take another pass through it. 

Thanks again!
